llvm/lib/Target/AArch64/GISel/AArch64LegalizerInfo.cpp

```diff
@@ -566,8 +553,10 @@ AArch64LegalizerInfo::AArch64LegalizerInfo(const AArch64Subtarget &ST)
                    Ty.getElementType() != SrcTy.getElementType();
           },
           0, 1)
-      .clampNumElements(0, v2s32, v4s32)
-      .clampMaxNumElements(1, s64, 2);
+      .clampNumElements(1, v4s16, v8s16)
+      .clampNumElements(1, v2s32, v4s32)
+      .clampMaxNumElements(1, s64, 2)
```

Comment:
```
.clampNumElements(1, v2s64, v2s64)
```

Reply:
There is only a max amount here, as it can't be a v1 vector. You could argue that keeping them more consistent is more readable, but only using clampNumElements is less legalizer rules.